src/inference_engines/vllm_engine.py

```diff
@@ -0,0 +1,154 @@
+import asyncio
+import json
+import os
+from io import IOBase
+from typing import BinaryIO, List, Union, get_args
+
+import torch
+from vllm import AsyncLLMEngine
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.sampling_params import SamplingParams
+
+from engine import Engine
```

Comment:
need the relative import here the way the repo's currently structured - `from .engine import Engine`